compiler/resolution/functionResolution.cpp

```diff
@@ -1723,11 +1724,19 @@ bool doCanDispatch(Type*     actualType,
         fn->hasFlag(FLAG_ALLOW_REF)))
     return true;
 
-  if (formalSym != NULL &&
-      formalSym->originalIntent == INTENT_INOUT &&
-      formalType->symbol->hasFlag(FLAG_REF)) {
-    actualType = actualType->getValType();
-    formalType = formalType->getValType();
+  if (formalSym != NULL && formalType->symbol->hasFlag(FLAG_REF)) {
+    if (formalSym->originalIntent == INTENT_INOUT) {
+      actualType = actualType->getValType();
+      formalType = formalType->getValType();
+    } else if (formalSym->intent == INTENT_CONST_REF) {
+      // ignore ref type when deciding what can pass to 'const ref'
```

Comment:
I'm guessing it's just easier to do this than to adjust code later down the line to consider `const ref` as a unique case?

Reply:
It seems that both this and code down the line needed adjustment.